src/RocketPoolEthTrigger.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Unlicensed
+pragma solidity 0.8.16;
+
+import "chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import "solmate/utils/FixedPointMathLib.sol";
+
+import "src/abstract/BaseTrigger.sol";
+import "src/interfaces/IRocketPoolOVMPriceOracle.sol";
+
+/**
+ * @notice A trigger contract that takes two addresses: a Chainlink oracle and a Rocket Pool OVM oracle.
+ * This trigger ensures the Chainlink oracle is always above the given price tolerance with respect to the
+ * Rocket Pool OVM oracle; the delta in prices can be equal to but not greater than the price tolerance.
+ * @dev Rocket Pool OVM oracles are specific to Optimism. Thus, this trigger should only be used on Optimism.
+ */
+contract RocketPoolEthTrigger is BaseTrigger {
+  using FixedPointMathLib for uint256;
+
+  uint256 internal constant WAD_DECIMALS = 18;
+  uint256 internal constant ZOC = 1e4;
+
+  /// @notice The market rate Chainlink oracle, assumed to be correct.
+  AggregatorV3Interface public immutable chainlinkOracle;
+
+  /// @notice The Rocket Pool OVM oracle, reporting the exchange rate from Rocket Pool on L1.
+  IRocketPoolOVMPriceOracle public immutable rocketPoolOracle;
+
+  /// @notice The maximum amount greater that the rocketPoolOracle price is than the chainlinkOracle price that is
+  /// allowed, expressed as a zoc.
+  /// For example, a 0.2e4 priceTolerance would mean the rocketPoolOracle price is
+  /// allowed to deviate from the chainlinkOracle price by up to +20%, but no more.
```

Comment:
I think we should only check how far above the rocket pool protocol rate is (not both sides). It doesn't really mean something back has happened if the chainlinkOracle price is far above - it's possible that the pool doesn't have enough ETH for withdrawals because of an imbalance with validators so it goes up.

Reply:
Checking only downside makes sense to me.

It would be nice to be consistent with the Chainlink triggers. Are they also only one-sided?

Reply:
chainlink triggers are both sided

Reply:
Got it.

Only downside make sense to me from protection buyer perspective.